### DISCUSSION TITLE
You have an error in your SQL syntax using INSERT-SET syntax

### DISCUSSION THREAD

**Discussion by cwoliveira** [sidorares/node-mysql2#2048] [Q&A]
I am in migration testing process from old "mysql" to "mysql2" npm package dependency.

So far, I cannot migrate directly due several "INSERT-SET" SQL statements used by applications.

1) Raw SQL running on database IDE - no error

`INSERT INTO teste1 SET nome='test 123';`

2) Query and parameters definition

```
query = "INSERT INTO teste1 SET ?";
params = {nome: "test 123456"};
```

3) Result from old mysql pacakge

```
sql result: OkPacket {
  fieldCount: 0,
  affectedRows: 1,
  insertId: 0,
  serverStatus: 2,
  warningCount: 0,
  message: '',
  protocol41: true,
  changedRows: 0
}
```

4) Result from mysql2 package

```
{
  text: `You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''{\\"nome\\":\\"test 123456\\"}'' at line 1`,
  sql: 'INSERT INTO teste1 SET ? - parameters:[{"nome":"test 123456"}]',
  fatal: false,
  errno: 1064,
  sqlState: '42000',
  code: 'ER_PARSE_ERROR'
}
```

**Comment by wellwelwel** [comment]
@cwoliveira, take a look at discussion #2037, may be useful 🙋🏻‍♂️
